pkg/ingester/limiter.go

```diff
@@ -144,7 +143,9 @@ func (l *Limiter) getShardSize(userID string) int {
 
 func (l *Limiter) getZonesCount() int {
 	if l.zoneAwarenessEnabled {
-		return util_math.Max(l.ring.ZonesCount(), 1)
+		// Series are replicated to at most replicationFactor zones, regardless of the number of zones in the ring
+		// Running with a number of zones < replicationFactor is unsupported
+		return util_math.Max(l.replicationFactor, 1)
```

Comment:
I'm not sure this change is correct. If you deploy Mimir in 5 zones, series are sharded between the 5 zones, but they're still replicated with RF=3. This means that each zone gets a lower number of series. When we convert a global limit to a local limit, I believe we have to take in account the actual number of zones, and not the RF.

Picking the example from the PR description.

> For example, consider:
> 
> Zone-aware cluster with 25 ingesters across 5 zones (5 ingesters per zone)
> - Replication factor = 3
> - -ingester.max-global-series-per-user = 10000
> - ingestion_tenant_shard_size = 0 (shuffle-sharding disabled)
> 
> So, the tenant's series should be replicated to all 5 ingesters in 3 different zones, with each of those ingesters having a local limit of 10000/5 = 2000 series.

The 10K input series are replicated to 30K series (RF=3). The 30K series are evenly distributed between ingesters across 5 zones, so 30K/5 = 6K series / zone. If we have 5 ingesters per zone, the local limit in each ingester should be 6K / 5 = 1200.

The current `Limiter` implementation (the version in `main`) does:

```
int((float64(globalLimit*l.replicationFactor) / float64(zonesCount)) / float64(ingestersInZoneCount))
```

So, it's `(10K * 3) / 5 zones / 5 ingester per zone = 1200`.

Looks correct to me.

Reply:
🤦 i got confused between sharding and replication, and should have checked how the tenant [subrings are actually calculated](https://github.com/grafana/dskit/blob/b9a439dedeb88a6ce8f90f6d7cb37e7fdd3330ab/ring/ring.go#L718-L729)